src/utils/btc.ts

```diff
@@ -0,0 +1,83 @@
+import * as ecc from "@bitcoin-js/tiny-secp256k1-asmjs";
+import { initEccLib, address, networks } from "bitcoinjs-lib";
+import { NO_COORD_PK_BYTE_LENGTH } from "../constants/keys";
+
+// Initialize elliptic curve library
+export function initBTCCurve() {
+  initEccLib(ecc);
+}
+
+/**
+ * Check whether the given address is a valid Bitcoin address.
+ *
+ * @param {string} btcAddress - The Bitcoin address to check.
+ * @param {object} network - The Bitcoin network (e.g., bitcoin.networks.bitcoin).
+ * @returns {boolean} - True if the address is valid, otherwise false.
+ */
+export const isValidBitcoinAddress = (
+  btcAddress: string,
+  network: networks.Network,
+): boolean => {
+  try {
+    return !!address.toOutputScript(btcAddress, network);
+  } catch (error) {
+    return false;
+  }
+};
+
+/**
+ * Check whether the given address is a Taproot address.
+ *
+ * @param {string} taprootAddress - The Bitcoin address to check.
+ * @param {object} network - The Bitcoin network (e.g., bitcoin.networks.bitcoin).
+ * @returns {boolean} - True if the address is a Taproot address, otherwise false.
+ */
+export const isTaproot = (taprootAddress: string, network: networks.Network): boolean => {
+  try {
+    const decoded = address.fromBech32(taprootAddress);
+    if (decoded.version !== 1) {
+      return false;
+    }
+    switch (network) {
+      case networks.bitcoin:
+        // Check if address statrts with "tb1p"
+        return taprootAddress.startsWith("bc1p");
+      case networks.testnet:
+        // signet, regtest and testnet taproot addresses start with "tb1p" or "sb1p"
+        return taprootAddress.startsWith("tb1p") || taprootAddress.startsWith("sb1p");
+      default:
+        return false;
+    }  
+  } catch (error) {
+    return false;
+  }
+};
+
+/**
+ * Check whether the given public key is valid.
+ *
+ * @param {string} publicKeyHex - The public key in hex to check. 
+ * It should be public key without the coordinate.
+ * @returns {boolean} - True if the public key is valid, otherwise false.
+ */
+export const isValidNoCordPublicKey = (publicKeyHex: string): boolean => {
+  return Buffer.from(publicKeyHex, "hex").length === NO_COORD_PK_BYTE_LENGTH;
```

Comment:
Wonder if this check is enough for checking whether something is a valid public key. Also, would the hex to byte conversion throw an error if the string is an invalid hex? If yes, is it expected for this function to throw an error?